designs/0031-deprecation-schedule.md

```diff
@@ -0,0 +1,254 @@
+must be implemented (thus 'deprecating' the use of that word as a identifier),
+new syntax which supersedes prior syntax, (such as `=` or `array[]`), or a
+similar **syntax-level** change. If supporting both the new and old way of doing
+things requires a 'hack' or difficult to maintain code, this is a prime
+candidate for this deprecation-removal process. If a change meaningfully changes
+the semantics of the language, it is not.
+
+>Current examples include the use of 'partial keywords' like offset,
+>which are not reserved words but act like them in some contexts, or multiple
+>syntaxes which lead to parsing difficulties, like the new array syntax. We would
+>like to not only improve the error messaging and implementation of these current
+>features, but also avoid introducing new issues of this variety in the long term.
+
+>Current non-examples include the ODE solver interface changes, which requires
+>changes to the user-defined function defining the ODE.
+
+Once it has been decided to deprecate a feature, the lexer, parser, or
+typechecker (whichever is most apropriate) should be updated to emit a
+compile-time warning notifying the user. This must contain the above 3 criteria:
+1. What is deprecated
+2. How to adjust your code
+3. When you need to adjust it by.
+
+The final bullet point should be decided during the review process with 3
+versions as a minimum. Many changes, such as reserving a new keyword, will not
+require longer periods, but syntax changes which the user may need time to
+adjust to could require longer time frames.
+
+This information must also be added to the documentation in the accompanying PR
+to the change.
+
+Finally, developers may wish to weigh in on how the implementation might look
+after the removal of the deprecated feature as part of the text of the pull
+request, if they can foresee code improvements which can occur at that time.
+
+## Code reviewers
+From a code reviewer standpoint, it is important to ask the following questions:
+1. Is this deprecation tied to a useful feature or noticeable improvement to the
+   underyling code (not just 'for the sake of it')?
+2. Does the new warning meet the criteria above and explain itself clearly?
+3. Is there a simple (preferably automated) way for users to accommodate this
+   change, e.g. through the `print-canonical` flag to stanc3 or a
+   find-and-replace feature of their text editor?
+4. Is the removal timeline appropriate for this change?
+
+## Releases
+Finally, during the release cycle an additional bullet point should be added to
+the release checklist to ensure that the deprecations slated for removal in the
+coming version have been properly handled. To accommodate this, the documentation
+page on deprecations should be sorted/categorized by removal version.
+
+Additionally, the release notes should contain a section on **Deprecations** and
+**Expired Deprecations** (alt. **Removals**) for each version. This should
```

Comment:
Let's just go with "Removed features" or "Breaking changes".  I think "expired deprecations" has one too many negative items and also isn't in the user's language.